### PR TITLE
feat: implement autocomplete search for cities

### LINKED ISSUE #8: Autocomplete for cities
At the moment, when you type in a city, it won't help you finish your entry with suggestions. A cool feature would be if it had a drop down menu that suggested locations similar to the name. For example, if I type in `Cambridge`, it could tell me `Cambridge, MA, USA` or `Cambridge, UK` to help users get the right answers!

### PR DESCRIPTION
**Pull request by IkehAkinyemi**
This proposed solution closes #8.

- This PR introduces a new component called CitySearchBar which replaces the input field in the app.
- The new component encapsulates the logic for fetching autocomplete city suggestions and handling user inputs for the city.
- The component uses the Geoapify Geocoding API service to fetch autocomplete suggestions for cities as the user types.
- This improves the overall structure of the app and makes it easier to maintain and modify in the future.

### CODE REVIEW

File: src/components/Autocomplete/index.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		}
+		setLocations(cleanedLocations);
+	}, [data]);
+
+	const searchAutocomplete = (city) => {
+		fetchAutocompleteResults(city);
+	};
+
+	const handleSelect = (label) => {
+		let city = label.name.split(',')[0];
```
Comment (pradumna-gautam) [inline src/components/Autocomplete/index.jsx:32]: Can we use 'const' here instead of 'let'?

**Review by pradumna-gautam — CHANGES_REQUESTED** [review]
@sumana2001 I was just curious, that yarn.lock has so many updates of the packages, is it really necessary to update so many packages in the repo?

File: src/App.js [hunk 2]
```diff
@@ -24,31 +32,38 @@ function App() {
...
+      <>
+        <img className="logo" src={logo} alt="MLH Prep Logo"></img>
+        <div>
+          <h2>Enter a city below 👇</h2>
+          <div className="input-container">
+            <Autocomplete setCity={setCity} />
+          </div>
+          <div className="Results">
+            {!isLoaded && <h2>Loading...</h2>}
+            {console.log(results)}
```
[the remark this answers is not in the document]
Comment (IkehAkinyemi) [reply src/App.js:51]: The change you referenced here wasn't made by me. Nevertheless, I will update it accordingly.

File: src/components/Autocomplete/index.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		setCity(city);
+	};
+
+	const searchBarStyling = {
```
[the remark this answers is not in the document]
Comment (IkehAkinyemi) [reply src/components/Autocomplete/index.jsx]: I have moved the styling for the searchBarStyling into App.css but the styles contained within the inputStyling variable can only be passed down as a styling prop to ReactSearchAutocomplete.

File: src/components/Autocomplete/index.jsx (hunk 1 above)
Comment (IkehAkinyemi) [reply src/components/Autocomplete/index.jsx]: Done

File: src/components/Autocomplete/index.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { ReactSearchAutocomplete } from 'react-search-autocomplete';
+import { useFetch } from '../../hooks/useFetch';
+
+const Autocomplete = ({ setCity, }) => {
```
[the remark this answers is not in the document]
Comment (IkehAkinyemi) [reply src/components/Autocomplete/index.jsx]: Done

**Comment by sumana2001** [comment]
@pradumna-gautam The changes to yarn.lock should only be committed if you added some new packages otherwise it's not required

**Review by pradumna-gautam — APPROVED** [review]

**Pull request merged**